### PR TITLE
Various Backend Tweaks

### PR DESCRIPTION
**Pull request by silent-po2**
This PR's primary purpose is to speed up test migrations for better CI flow and for ease of running tests locally. It additionally fixes some rare bugs that can cause a crash when flagging/approving data sets.

### CODE REVIEW

**Comment by Arrow105** [comment]
The bug reported by SonarCloud is a false positive.

File: Server/src/models/DatasetModels/DatasetApprovalModel.ts [hunk 1]
```diff
@@ -70,9 +71,12 @@ export class DatasetApprovalModel {
     }
 
     async updateDatasetComments(datasetId: number, datasetCommentsToAppend?: string) {
-        if (datasetCommentsToAppend !== null && datasetCommentsToAppend !== undefined) {
+        if (datasetCommentsToAppend !== undefined && datasetCommentsToAppend !== "undefined" && datasetCommentsToAppend !== null) {
```
Comment (ratulkai93) [inline Server/src/models/DatasetModels/DatasetApprovalModel.ts:74]: nice check

File: Server/src/models/entities/Dataset.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ import { Accounts } from './Accounts';
  * and has multiple composition references for fuel, oxidizer, and diluent
  */
 @Entity()
-export class Dataset {
+export class Dataset extends BaseEntity {
```
Comment (ratulkai93) [inline Server/src/models/entities/Dataset.ts:16]: just wondering what is purpose of adding the BaseEntity?

File: Server/src/tests/services/DataSetService.spec.ts [hunk 3]
```diff
@@ -234,13 +234,13 @@ describe('data set service test', () => {
...
-  //   expect(response.statusCode).toEqual(200);
-  //   done()
-  // });
+  test('Asks for all unapproved data sets, expects at least one data set', async done => {
+    let response = await retrieveDataObject.getUnapprovedAllDatasets()
+    let arrayOfData = response.message as IApprovalDatasetModel[]
+    expect(arrayOfData[0]).not.toBeUndefined();
+    expect(response.statusCode).toEqual(200);
+    done()
+  });
```
Comment (ratulkai93) [inline Server/src/tests/services/DataSetService.spec.ts:243]: tests looking great. good job !

**Review by ratulkai93 — APPROVED** [review]
Amazing job. Thanks for your hard work!

File: Server/src/models/entities/Dataset.ts (hunk 2 above)
Comment (silent-po2) [reply Server/src/models/entities/Dataset.ts:16]: It allows usage of more functions, like '.save' from the entity itself.

Comment (ratulkai93) [reply Server/src/models/entities/Dataset.ts:16]: i see

File: Server/src/models/AuthenticationModel.ts [hunk 4]
```diff
@@ -27,7 +24,7 @@ export class AuthenticationModel {
     signUpInformation.orcID = signUpInfo.orcID;
     signUpInformation.organizationName = signUpInfo.organizationName;
     signUpInformation.admin = signUpInfo.isAdmin;
-    await connection.manager.save(signUpInformation);
+    await Accounts.save(signUpInformation);
```
Comment (ds161833) [inline Server/src/models/AuthenticationModel.ts:27]: nice

File: Server/src/models/DatasetModels/DatasetApprovalModel.ts (hunk 1 above)
Comment (ds161833) [inline Server/src/models/DatasetModels/DatasetApprovalModel.ts:74]: i dont think its possible for it to be a string of "undefined", just if(datasetCommentToAppend) should suffice

File: Server/src/models/DatasetModels/DatasetApprovalModel.ts [hunk 5]
```diff
@@ -70,9 +71,12 @@ export class DatasetApprovalModel {
     }
 
     async updateDatasetComments(datasetId: number, datasetCommentsToAppend?: string) {
-        if (datasetCommentsToAppend !== null && datasetCommentsToAppend !== undefined) {
+        if (datasetCommentsToAppend !== undefined && datasetCommentsToAppend !== "undefined" && datasetCommentsToAppend !== null) {
             let oldComment = await this.selectDatasetCommentQuery(datasetId)
-            let newComment = oldComment.comments.concat(" " + datasetCommentsToAppend)
+            let newComment = datasetCommentsToAppend
+            if (oldComment.comments !== null) {
```
Comment (ds161833) [inline Server/src/models/DatasetModels/DatasetApprovalModel.ts]: same here if (oldComment.comments) <- should do the job

File: Server/src/models/entities/Accounts.ts [hunk 6]
```diff
@@ -1,9 +1,9 @@
-import { Entity, Column, PrimaryGeneratedColumn, CreateDateColumn, UpdateDateColumn, JoinTable, ManyToMany, Connection } from "typeorm";
+import { Entity, Column, PrimaryGeneratedColumn, CreateDateColumn, UpdateDateColumn, JoinTable, ManyToMany, Connection, BaseEntity } from "typeorm";
 import { Dataset } from "./Dataset";
 
 
 @Entity()
-export class Accounts {
+export class Accounts extends BaseEntity {
```
Comment (ds161833) [inline Server/src/models/entities/Accounts.ts:6]: nice

**Review by ds161833 — APPROVED** [review]

File: Server/src/models/DatasetModels/DatasetApprovalModel.ts (hunk 1 above)
Comment (silent-po2) [reply Server/src/models/DatasetModels/DatasetApprovalModel.ts:74]: It is in fact possible to get a string of "undefinied" and that was the primary issue to solve.

File: Server/src/models/DatasetModels/DatasetApprovalModel.ts (hunk 5 above)
Comment (silent-po2) [reply Server/src/models/DatasetModels/DatasetApprovalModel.ts]: This might work, will check.

**Pull request merged**